### DISCUSSION TITLE
Import assertions with swc enabled

### DISCUSSION THREAD

**Discussion by leonitousconforti** [TypeStrong/ts-node#1810] [Q&A] [answered]
I am having difficulty using import assertions with ts-node when swc is enabled. Import assertions work just fine when swc is disabled, but not when swc is enabled. Import assertions seem to be behind a config flag for swc per https://github.com/swc-project/swc/issues/2802.

> The import assertions should stay intact if the target environment is set to esnext, module system is esm and importAssertions in the config is set to true.

but I can't find any way to set importAssertions to true. I can see that ts-node creates its own swcconfig from the tsconfig [here](https://github.com/TypeStrong/ts-node/blob/f34d87479b979fc2a9442f3f576893dcf6b620a8/src/transpilers/swc.ts#L66-L149) and I don't really see anyway to override or add to these options.

Am I missing something? Should this be enabled by default? I would be happy to move this to a new issue or create a pr if that would be more appropriate.

**Comment by cspotcode** [comment, marked as answer]
Looks like you've found a bug, thanks for reporting.  Not only do we need to specify `importAssertions`, but swc also requires `experimental: {keepImportAssertions: true}`

I'm guessing that's why this bug exists, because SWC didn't support this when we first added support.

I've created a bug to track #1817, and I already have a PR in progress to fix #1802.  Thanks again for reporting this.

**Reply by leonitousconforti** [reply]
Thanks for taking the time to help, I appreciate it!
